infra/build/functions/request_build_test.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Unit tests for Cloud Function request builds which builds projects."""
+
+import datetime
+import os
+import subprocess
+import threading
+import unittest
+
+import requests
+
+from google.cloud import ndb
+
+import mock
+
+from request_build import get_build_steps
+from datastore_entities import Project
+
+_EMULATOR_TIMEOUT = 20
+_DATASTORE_READY_INDICATOR = b'is now running'
+_DATASTORE_EMULATOR_PORT = 8432
+_TEST_PROJECT_ID = 'test-project'
+
+
+def start_datastore_emulator():
```

Comment:
Please put the common datastore emulator code into a test_utils.py module and reuse everywhere.